lib/core/test/unit/Cardano/Wallet/Primitive/CoinSelection/LargestFirstSpec.hs

```diff
@@ -133,6 +133,26 @@ spec = do
                 , txOutputs = 40 :| [1,1,1]
                 }
 
+        coinSelectionUnitTest
+            largestFirst
+            "enough coins, fragmented enough, but one output depletes all inputs"
+            (Left ErrInputsDepleted)
+            $ CoinSelectionFixture
+                { maxNumOfInputs = 100
+                , utxoInputs = [12,20,17]
+                , txOutputs = 40 :| [1]
+                }
+
+        coinSelectionUnitTest
```

Comment:
this is one of very interesting cases. Occassionally, `random` is better than this if for `41` it chooses `[20,20,5]` and for `6` output `10`. **largestFirst** gives `ErrInputsDepleted` here deterministically. `random` can give non-error. If the dices are against us here in case of `random` we have fallback to `largestFirst`. Saying that, at this  moment, I see only one additional property - there are cases when `largestFirst` give deterministically error, but in such cases, `random` can in non-deterministic way choose valid inputs. The question for me is if we want to add such a property

Reply:
It'd be hard to express such property actually :thinking: ... Have you "rough" idea in mind already of what this could look like ?

Reply:
agree here. it is very specific case, but I believe it is going to pop up someday. We need additional coin to select to cover one output, and there are two coins left, both can cover the output. But we have also last output to cover, and here only larger one coin can cover it. in `largestFirst` we are doing always this bad choice ending up in error. In case of `random` we have 50% chance to do it right (if we are already facing this last random pick). And here `random` shows its precedence. So in order to provide property like this we need to prepare very specific Coin Selection generator emulating such last choice....